Star Seeker/Views/Compositions/MovementController/JoystickMovementController.swift

```diff
@@ -1,28 +1,41 @@
 import SpriteKit
 
 class JoystickMovementController : MovementController {
+    /** Specifies how large the joystick knob will be */
+    let bSize   = UIConfig.SquareSizes.mini + 10
+    /** Convenience class-exclusive variable, which gets its value by deriving horizontal impulse value from GameConfig */
+    let hImpls   : CGFloat = GameConfig.lateralImpulse
+    /** Convenience class-exclusive variable, which gets its value by deriving vertical impulse value from GameConfig */
+    let vImpls   : CGFloat = GameConfig.elevationalImpulse
+    /** Convenience class-exclusive variable, which gets its value by deriving joystick safe area distance value from GameConfig */
+    let joystickSafeArea   : CGFloat = GameConfig.joystickSafeArea
+    /** Convenience class-exclusive variable, which gets its value by deriving joystick max area distance value from GameConfig */
+    let joystickMaxDistance   : CGFloat = GameConfig.joystickMaxDistance
     
     override var target: Player {
         didSet {
             self.directionIndicator?.removeFromParent()
             target.addChild(self.directionIndicator!)
         }
     }
-    var directionIndicator : SKSpriteNode?
     
-    override init ( controls target : Player ) {        
-        self.hImpls = GameConfig.lateralImpulse
-        self.vImpls = GameConfig.elevationalImpulse
-        
-        self.directionIndicator = JoystickMovementController.defaultDirectionIndicator(to: target)
-        
+    var directionIndicator : SKShapeNode?
```

Comment:
directionIndicator now means the dashed line which marks the player's future trajectory?

Reply:
correct